### DISCUSSION TITLE
PCB with integrated Power supply

### DISCUSSION THREAD

**Discussion by alexseuf** [hoylabs/OpenDTU-OnBattery#560] [Show and tell]
Hallo,
konnte die Finger nicht still lassen und habe nun doch eine eigene Platine entworfern

siehe:
https://www.akkudoktor.net/forum/migrated-forums-balkonsolar/opendtu-on-battery-platine-mit-maximal-ausbau-dc-dc-und-jk-bms-stecker/#post-171934

- Vorteil insb. mit JK-BMS: nur ein 1zu1 Kabel zum BMS für Power und Daten und eine 1zu1 Kabel zum Victron MPPT
- integriertes 60V netzteil
- Verschiedene BMS können über Jumper ausgewählt werden
- Alle Anschlüsse Steckbar; möglichst mit original 1zu1 Kabel
- ESP32 & NRF24 können selbst je nach Anwendungsfall in der Variante mit oder ohne externe Antenne bestückt werden
- CanBus & Optokoppler für Huawei R4850G2 power supply
- Leds für RX & TX der Batterie Kommunikation und für RX zum MPPT
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/cb740410-6ddc-432e-8e35-6c93bc9dce29)

Stromlaufplan:
[Schematic_OPenDTU onBattery_2023-12-23.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13759410/Schematic_OPenDTU.onBattery_2023-12-23.pdf)

Freue mich über Anregungen, Hinweise und Kritik

**Comment by Manos1966** [comment]
> ESP32 & NRF24 können selbst je nach Anwendungsfall in der Variante mit oder ohne externe Antenne bestückt werden

Eine Frage @alexseuf Kann ich an der Stelle des NRF24, ein CMT2300A Funkmodul nutzen?
![Ebyte Funkmodul CMT2300A 868915MHz Breakoutboard](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ba27b740-e4e4-44ac-92bb-7d8bf49a47eb)

**Reply by alexseuf** [reply]
genau ist so vorgesehen(elektrisch & mechanisch), hättest du interesse das zu testen?

**Reply by Manos1966** [reply]
ich habe dir bereits im Akkudoktor Forum beantwortet 😉 

Victron, Pylontech, Huawei R4580G kann ich testen.
Aber JK-BMS habe ich leider nicht.

**Reply by alexseuf** [reply]
JK BMS,  und smart shunt habe ich, super dann sollte ja das meiste abgedeckt sein.

**Reply by Manos1966** [reply]
...und ich kann nur NRF24 testen.
(Ich habe z.Zeit keine Hoymiles die ein CMT2300A Funkmodul nutzt 😞 )

**Comment by swingstate** [comment]
Schönes Ding. 👌 Dann kann ich mir die Arbeit an weiteren Platinen sparen.

**Reply by alexseuf** [reply]
@swingstate Kannst Du mir noch mal kurz die farbzuordnung der drei LEDs bei dir nennen,  damit ich diese identisch machen kann, wenn das für dich okay ist, evtl auch die gewählten Vorwiderstände? Hast du Erfahrung mit esd Schutzmaßnahmen auf den Datenleitungen, bzw. Wie würdest du meinen Versuch bewerten?
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/7af7fc2d-34f9-4525-8314-1afbf6ff5861)

**Reply by ottelo9** [reply]
Solche Bidirektionalen ESD Schutzdioden verwende ich bei meinen Platinen für die Arbeit ebenfalls. Wenn man diese korrekt bzgl. den Spannungspegeln (Breakdown Voltage !) ausgewählt hat, bieten sie einen guten Schutz gegen ESD. Hatte in der Vergangenheit ab und zu Probleme mit defekten Mosfets oder Endstufen. Nachdem ich dann ESD Dioden (und Snubber Netzwerke) eindesigned hatte, gabs nie wieder Probleme. Bei Datenleitungen muss man natürlich aufpassen, dass die Kapazität der Dioden nicht zu hoch ist. Sonst werden die Signale bei hohen Frequenzen gedämpft.

Würth Elektronik hat da auch immer gute EMV/ESD Beispiele (und Kits) für USB und RS485. https://www.we-online.com/de/components/products/EMC_FILTER_BAGS_STICKS_RS_485_PROFIBUS_FILTERSTICK

**Comment by alexseuf** [comment]
so...  habe jetzt die Platine finalisiert:
[PCB_PCB_OPenDTU onBattery_2023-12-25-4.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13767401/PCB_PCB_OPenDTU.onBattery_2023-12-25-4.pdf)
[Schematic_OPenDTU onBattery_2023-12-25.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13767402/Schematic_OPenDTU.onBattery_2023-12-25.pdf)


wenn niemenden mehr was auffällt würde ich morgen so bestellen.

Durch die Mehrfarbigen LEDs und die LED-Treiber sind die Bestückungskosten noch etwas gestiegen, komme jetzt auf 29€ pro platine (plus Versand in D)

Wer hat noch interesse an einem Muster?

Bestückumfang:
![WhatsApp Bild 2023-12-25 um 22 23 32_d828cf64](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/06da7bed-6e10-4950-ad03-e54ce2a1ecff)

Wer den MCP2515 (Huawei CAN) oder den ADUm1201 (für Victron Smart Shunt) braucht, dem löte ich einen aus meinem Fundus drauf.. aber der muss dann auch Feedback von den Tests geben :-)

ein Gehäuse ist auch in Arbeit
![WhatsApp Bild 2023-12-25 um 16 55 40_bb392a15](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/9cea587c-914b-4dcf-90ad-2316603a8e97)
https://drive.google.com/file/d/1-FtYoj1LDjfDwLTJWii7A52UlbZ_ePGo/view?pli=1

**Reply by Manos1966** [reply]
Wow! Das war schnell 👍
Ich hätte gerne eine Platine, aber mit Huawei Canbus bitte.

**Comment by alexseuf** [comment]
Hallo bin gerade bei der Inbetriebnahme der Platine, bisher funktioniert alles.
Jedoch bekomme ich die Firmware nicht dazu den Huawei Power Pin auf GND zu schalten, der ist bei mir immer auf 3V3.

Mein Versuch mittels MQTT-Explorer das huawei/cmd/mode .Register zu überschrieben brachte auch keinen Erfolg...
![27779-grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/49fa903f-9b13-4f2d-a366-be9e238c29ea)
![27780-WhatsApp-Bild-2024-01-14-um-12-55-52ff487227](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/1e9df422-cc1c-4bba-873a-66eff433ae4c)

kann mir da jeamand Nachhilfe geben?

p.S.
den aktuellen stand der Inbetriebnahme kann man hier verfolgen:
https://www.akkudoktor.net/forum/migrated-forums-balkonsolar/opendtu-on-battery-platine-mit-maximal-ausbau-dc-dc-und-jk-bms-stecker/paged/3/#post-176680

**Reply by alexseuf** [reply]
hat sich erledigt, der Fehler war das ich die Daten als raw senden muss

**Comment by alexseuf** [comment]
Die Platine ist fertig getestet, wer interesse hat eine Platine im Live Betrieb zu testen, der meldet sich am besten per PN im akkudoctor Forum: 
https://www.akkudoktor.net/forum/migrated-forums-balkonsolar/opendtu-on-battery-platine-mit-maximal-ausbau-dc-dc-und-jk-bms-stecker/paged/4/#post-177458
Kostenpunkt 30€ €+4,5€Versand (damit gebe ich den Preis den ich gezahlt habe 1zu1 weiter) 

hier mal ein Bild von dem Signal- Verdrahtungsaufwand für ein Balkonkraftwerk mit

    Hoymiles-WR,
    victron MPPT,
    und Batterie mit JK-BMS-Batterie

über das 1zu1 Kabel zur BMS kommt auch die Spannungsversorung für die openDTU, also nur zwei fertige Käbelchen notwendig!
![WhatsApp Bild 2024-01-16 um 20 19 36_072be33d](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/48165001/f49b9fc5-1088-4dce-97ed-f04a1b29cf41)

**Reply by alexseuf** [reply]
Hier noch das aktuelle Pin-Mapping:

```
[
    {
        "name": "AlexDTU with Battery + Victron MPPT + HUAWEI +NRF24 + LED",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "victron": {
            "rx": 22,
            "tx": -1
        },
        "battery": {
            "rx": 27,
            "rxen": 33,
            "tx": 14,
            "txen": 32
        },
        "led": {
            "led0": 17,
            "led1": 1
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        },
        "huawei": {
            "miso": 12,
            "mosi": 13,
            "clk": 26,
            "irq": 25,
            "power": 21,
            "cs": 15
        }
    }
]
```

**Comment by rs3hc** [comment]
Moin Alex.
Kann man die Platine bei dir auch fertig aufgebaut kaufen?

LG Robert

**Reply by alexseuf** [reply]
Kannst du mir über das Akkudoktor Forum eine Private Nachricht mit deinen Kontaktdaten senden? Ich kenne keine Möglichkeit hier private Daten auszutauschen.
